public/components/custom_panels/helpers/utils.tsx

```diff
@@ -384,6 +391,183 @@ export const renderCatalogVisualization = async ({
+    },
+  };
+};
+
+export const fetchAggregatedBinCount = (
+  minimumBound: string,
+  maximumBound: string,
+  startTime: string,
+  endTime: string,
+  documentName: string,
+  http: CoreStart['http'],
+  selectedOtelIndex: string,
+  setIsError: React.Dispatch<React.SetStateAction<VizContainerError>>,
+  setIsLoading: React.Dispatch<React.SetStateAction<boolean>>
+) => async () => {
+  return http
+    .post(`${OBSERVABILITY_BASE}/metrics/otel/aggregatedBinCount`, {
+      body: JSON.stringify({
+        min: minimumBound,
+        max: maximumBound,
+        startTime,
+        endTime,
+        documentName,
+        index: selectedOtelIndex,
+      }),
+    })
+    .catch((error: Error) => {
+      const errorMessage = JSON.parse(error.body.message);
+      setIsError({
+        errorMessage: errorMessage.error.reason || 'Issue in fetching bucket count',
+        errorDetails: errorMessage.error.details,
+      });
+      console.error(error.body);
+    })
+    .finally(() => {
+      setIsLoading(false);
+    });
+};
+
+export const fetchSampleOTDocument = (
+  selectedOtelIndex: string,
+  http: CoreStart['http'],
+  documentName: string
+) => async () => {
+  return http
+    .post(`${OBSERVABILITY_BASE}/metrics/otel/sampleDocument`, {
+      body: JSON.stringify({
+        documentName,
+        index: selectedOtelIndex,
+      }),
+    })
+    .catch((error) => console.error(error));
+};
+
+export const extractIndexAndDocumentName = (metricString: string): [string, string] | null => {
+  const pattern = /\[Otel Metric\]\s(\S+?-\S+?)\.(\S+)/;
+  const match = metricString.match(pattern);
+
+  if (match) {
+    const index = match[1];
+    const documentName = match[2];
+    return [index, documentName];
+  } else {
+    return null;
+  }
+};
+
+export const renderOpenTelemetryVisualization = async ({
+  startTime,
+  endTime,
+  setVisualizationTitle,
+  setVisualizationType,
+  setVisualizationData,
+  setVisualizationMetaData,
+  setIsLoading,
+  setIsError,
+  visualization,
+}: {
+  startTime: string;
+  endTime: string;
+  setVisualizationTitle: React.Dispatch<React.SetStateAction<string>>;
+  setVisualizationType: React.Dispatch<React.SetStateAction<string>>;
+  setVisualizationData: React.Dispatch<React.SetStateAction<Plotly.Data[]>>;
+  setVisualizationMetaData: React.Dispatch<React.SetStateAction<undefined>>;
+  setIsLoading: React.Dispatch<React.SetStateAction<boolean>>;
+  setIsError: React.Dispatch<React.SetStateAction<VizContainerError>>;
+  visualization: any;
+}) => {
+  setIsLoading(true);
+  setIsError({} as VizContainerError);
+
+  const { http } = coreRefs;
+  const visualizationType = 'bar';
+  let index = visualization?.index;
+  let documentName = visualization?.name;
+
+  if (index === undefined) {
+    const indexAndDocumentName = extractIndexAndDocumentName(visualization.name);
+    index = indexAndDocumentName[0];
+    documentName = indexAndDocumentName[1];
```

Comment:
null pointer exception if `extractIndexAndDocumentName` returned null?

Reply:
handled it